### PR TITLE
fix: [icx-proxy] Sort aliases from longest to shortest

### PR DESCRIPTION
**Pull request by ericswanson-dfinity**
Fixes https://github.com/dfinity/agent-rs/issues/153

### CODE REVIEW

File: icx-proxy/src/config/canister_dns_config.rs [hunk 1]
```diff
@@ -11,9 +12,11 @@ impl CanisterDnsConfig {
     /// Create a CanisterDnsConfig instance from command-line configuration.
     /// dns_aliases: 0 or more entries of the form of dns.alias:canister-id
     pub fn new(dns_aliases: &[String]) -> anyhow::Result<CanisterDnsConfig> {
+        let mut dns_aliases: Vec<String> = dns_aliases.iter().map(String::clone).collect();
+        dns_aliases.sort_by_key(|s| Reverse(s.len()));
         let dns_aliases = dns_aliases
             .iter()
-            .map(|alias| DnsAlias::new(alias))
+            .map(|alias| DnsAlias::new(&alias))
             .collect::<anyhow::Result<Vec<DnsAlias>>>()?;
```
Comment (hansl) [inline icx-proxy/src/config/canister_dns_config.rs:20]: I with we wouldn't have to collect twice, but oh well... It's just at startup anyway.

File: icx-proxy/src/config/canister_dns_config.rs [hunk 2]
```diff
@@ -143,6 +146,86 @@ mod tests {
...
+        );
+    }
+
+    #[test]
+    fn searches_longest_to_shortest_not_alpha_reversed() {
+        // Similar to searches_longest_to_shortest, but make sure we
+        // don't happen to get there by sorting alphabetically/reversed
+        let dns_aliases = parse_dns_aliases(vec![
+            "x.a.c:r7inp-6aaaa-aaaaa-aaabq-cai",
+            "a.c:rrkah-fqaaa-aaaaa-aaaaq-cai",
```
Comment (hansl) [inline icx-proxy/src/config/canister_dns_config.rs:215]: Can you reverse the order you pass them? If it's not sorted this test would still pass, right?

**Review by hansl — CHANGES_REQUESTED** [review]
One non blocking, and another to make the test better. Otherwise LGTM

File: icx-proxy/src/config/canister_dns_config.rs (hunk 2 above)
Comment (ericswanson-dfinity) [reply icx-proxy/src/config/canister_dns_config.rs:215]: The test above passes them in the reversed order from this one.  Does that cover the case you were wanting to?

File: icx-proxy/src/config/canister_dns_config.rs (hunk 1 above)
Comment (ericswanson-dfinity) [reply icx-proxy/src/config/canister_dns_config.rs:20]: Me too, but I couldn't sort in between `.iter()` and `.collect()`..  I have an idea though

File: icx-proxy/src/config/canister_dns_config.rs (hunk 2 above)
Comment (ericswanson-dfinity) [reply icx-proxy/src/config/canister_dns_config.rs:214]: Yeah, I missed something on this one.. will adjust

**Review by hansl — APPROVED** [review]

**Pull request merged**